src/models/ml_trainer.py

```
@@ -0,0 +1,331 @@
+    """
+    p = Path(path)
+    p.parent.mkdir(parents=True, exist_ok=True)
+    with p.open("wb") as f:
+        pickle.dump(model, f)
+    logger.info("[ML] saved → %s", p)
+
+
+def load_model(path: str | Path) -> Any:
+    """pickle 로 모델 로드. save_model 의 짝."""
+    with Path(path).open("rb") as f:
+        return pickle.load(f)
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)